src/main/java/blusunrize/immersiveengineering/api/excavator/ExcavatorHandler.java

```diff
@@ -0,0 +1,199 @@
+		MineralWorldInfo worldInfo = MINERAL_INFO_CACHE.get(columnPos);
+		if(worldInfo==null)
+		{
+			List<Pair<MineralVein, Double>> inVeins = new ArrayList<>();
+			double totalSaturation = 0;
+			MineralMix mix = null;
+			// Iterate all known veins
+			for(MineralVein vein : MINERAL_VEIN_LIST.get(world.getDimension().getType()))
+			{
+				int dX = vein.getPos().x-columnPos.x;
+				int dZ = vein.getPos().z-columnPos.z;
+				int d = dX*dX+dZ*dZ;
+				double rSq = vein.getRadius()*vein.getRadius();
+				if(d < rSq)
+				{
+					double saturation = 1-(d/rSq);
+					inVeins.add(Pair.of(vein, saturation));
+					totalSaturation += saturation;
+				}
+			}
+			final double finalTotalSaturation = totalSaturation;
+			worldInfo = new MineralWorldInfo(inVeins.stream()
+					.map(pair -> Pair.of(pair.getLeft(), (int)(pair.getRight()/finalTotalSaturation*1000)))
+					.collect(Collectors.toList())
+			);
+			MINERAL_INFO_CACHE.put(columnPos, worldInfo);
+		}
+		return worldInfo;
+	}
+
+	public static void generatePotentialVein(DimensionType dimension, ChunkPos chunkpos, Random rand)
+	{
+		int xStart = chunkpos.getXStart();
+		int zStart = chunkpos.getZStart();
+		double d0 = 0.0625D;
+		boolean containsVein = false;
+		ColumnPos pos = null;
+		iteratechunk:
+		for(int xx = 0; xx < 16; ++xx)
+			for(int zz = 0; zz < 16; ++zz)
+			{
+				double noise = noiseGenerator.noiseAt((xStart+xx)*d0, (zStart+zz)*d0, d0, xx*d0);
+				double chance = Math.abs(noise)/.55;
+				if(chance > mineralNoiseThreshold)
+				{
+					containsVein = true;
+					pos = new ColumnPos(xStart+xx, zStart+zz);
+					break iteratechunk;
+				}
+			}
+		if(containsVein)
+		{
+			ColumnPos finalPos = pos;
+			int radius = 12+rand.nextInt(32);
+			int radiusSq = radius*radius;
+			boolean crossover = MINERAL_VEIN_LIST.get(dimension).stream().anyMatch(vein -> {
+				int dx = vein.getPos().x-finalPos.x;
+				int dz = vein.getPos().z-finalPos.z;
+				return dx*dx+dz*dz < vein.getRadius()*vein.getRadius()||dx*dx+dz*dz < radiusSq;
+			});
+			if(!crossover)
+			{
+				MineralMix mineralMix = null;
+				MineralSelection selection = new MineralSelection(dimension, 2);
+				if(selection.getTotalWeight() > 0)
+				{
+					int weight = selection.getRandomWeight(rand);
+					for(MineralMix e : selection.getMinerals())
+					{
+						weight -= e.weight;
+						if(weight < 0)
+						{
+							mineralMix = e;
+							break;
+						}
+					}
+				}
+				if(mineralMix!=null)
+				{
+					MineralVein vein = new MineralVein(pos, mineralMix, radius);
+					// generate initial depletion
+					if(initialVeinDepletion > 0)
+						vein.setDepletion((int)(mineralVeinYield*(rand.nextDouble()*initialVeinDepletion)));
+					MINERAL_VEIN_LIST.put(dimension, vein);
+					IESaveData.setDirty();
+				}
+			}
+		}
+	}
+
+	public static class MineralSelection
+	{
+		private final int totalWeight;
+		private final Set<MineralMix> validMinerals;
+
+		public MineralSelection(DimensionType dimension, int radius)
```

Comment:
What is the `radius`-parameter needed for?

Reply:
Nothing anymore! Legacy stuff I forgot to remove :D